src/go/cmd/http-relay-server/server/broker.go

```diff
@@ -259,9 +263,17 @@ func (r *broker) SendResponse(resp *pb.HttpResponse) error {
 
 	// Writing to this channel will notify consumers which are waiting for data
 	// on the channel returned by RelayRequest().
-	pr.responseStream <- resp
-
+	select {
+	case pr.responseStream <- resp:
```

Comment:
In the latest solution, I lock `sendMutex` instead of the main broker mutex. There is a `sendMutex` per connection per connection. `ReapInactiveRequests` may still deadlock though, ~but at least that won't degrade the perf as much~. Maybe I can use `TryLock()`, and if I can't obtain a lock, and if I haven't close `markReap`, then I close it and wait for the next trigger to cleanup (if it still exist).

Reply:
This seems safe as we close markReap before locking. Everything else shouldn't take long (metrics, logging, closing responseStream).